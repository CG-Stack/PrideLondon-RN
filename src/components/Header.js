// @flow
import React from "react";
import { StyleSheet, View } from "react-native";
import type { Node } from "react";
import SafeAreaView from "react-native-safe-area-view";
import ContentPadding from "./ContentPadding";
import IconButton from "./IconButton";
import Text from "./Text";
import chevronLeftWhite from "../../assets/images/chevron-left-white.png";
import { lightNavyBlueColor } from "../constants/colors";
import text from "../constants/text";

type Props = {
  onBack?: () => void,
  title?: string,
  rightElement?: Node
};

const renderBackButton = onBack => (
  <IconButton
    accessibilityLabel={text.backButtonAccessibilityLabel}
    onPress={onBack}
    source={chevronLeftWhite}
  />
);

const renderTitle = (title: string) => (
  <Text type="h2" color="whiteColor">
    {title}
  </Text>
);

const Header = ({ onBack, title, rightElement }: Props) => {
  const leftElement = onBack ? renderBackButton(onBack) : null;
  const titleElement = title ? renderTitle(title) : null;
  return (
    <SafeAreaView
      accessibilityTraits={["header"]}
      forceInset={{ top: "always" }}
      style={{ backgroundColor: lightNavyBlueColor }}
    >
      <ContentPadding style={styles.headerContent}>
        {leftElement || <View style={styles.phantomIcon} />}
        {titleElement}
        {rightElement || <View style={styles.phantomIcon} />}
      </ContentPadding>
    </SafeAreaView>
  );
};

Header.defaultProps = {
  title: undefined,
  onBack: undefined,
  rightElement: undefined
};

const styles = StyleSheet.create({
  headerContent: {
    height: 48,
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    alignSelf: "center",
    maxWidth: 440,
    width: "100%"
  },
  phantomIcon: {
    width: 48,
    height: 48
  }
});

export default Header;
