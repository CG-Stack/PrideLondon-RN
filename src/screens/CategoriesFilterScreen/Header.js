// @flow
import React from "react";
import { View, TouchableOpacity, StyleSheet } from "react-native";
import Text from "../../components/Text";
import text from "../../constants/text";
import { whiteColor } from "../../constants/colors";
import CategoriesPills from "../../components/CategoriesPills";

type Props = {
  onClose?: Function,
  onClearAll?: Function,
  selectedCategories: Set<string>
};

type ActionButtonProps = {
  label: string,
  onPress?: Function
};

const ActionButton = ({ label, onPress }: ActionButtonProps) => (
  <TouchableOpacity style={styles.actionButton} onPress={onPress}>
    <Text type="h4" style={styles.actionButtonText}>
      {label}
    </Text>
  </TouchableOpacity>
);

ActionButton.defaultProps = {
  onPress: () => {}
};

const Header = ({ onClose, onClearAll, selectedCategories }: Props) => (
  <View>
    <View style={styles.actionButtons}>
      <ActionButton label={text.cancel} onPress={onClose} />
      {selectedCategories.size > 0 && (
        <ActionButton label={text.clearAll} onPress={onClearAll} />
      )}
    </View>
    <Text type="h1" style={styles.filterTitle}>
      {text.filterTitle}
    </Text>
    <CategoriesPills
      style={styles.categoriesPills}
      selectedCategories={selectedCategories}
    />
  </View>
);

Header.defaultProps = {
  onClose: () => {},
  onClearAll: () => {}
};

const styles = StyleSheet.create({
  actionButtons: {
    width: "100%",
    flexDirection: "row",
    justifyContent: "space-between",
    marginTop: 10
  },
  actionButton: {
    width: 60
  },
  actionButtonText: {
    color: whiteColor
  },
  categoriesPills: {
    marginBottom: 16
  },
  filterTitle: {
    color: whiteColor,
    marginTop: 14,
    marginBottom: 6
  }
});

export default Header;
