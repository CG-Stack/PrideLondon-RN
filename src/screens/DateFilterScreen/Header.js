// @flow
import React from "react";
import { View } from "react-native";
import CommonHeader from "../../components/Header";
import ActionButton from "../../components/ActionButton";

type Props = {
  onCancel: () => void,
  onReset: () => void,
  dateRange: string,
  title: string,
  titleLabel: string
};

const Header = ({ onCancel, onReset, dateRange, title, titleLabel }: Props) => (
  <View>
    <CommonHeader
      leftElement={<CommonHeader.BackButton onPress={onCancel} />}
      title={title}
      titleLabel={titleLabel}
      titleAccesible
      titleAccessibilityTraits={["header"]}
      titleAccessibilityLiveRegion="polite"
      rightElement={
        dateRange && <ActionButton label="Reset" onPress={onReset} />
      }
    />
  </View>
);

export default Header;
