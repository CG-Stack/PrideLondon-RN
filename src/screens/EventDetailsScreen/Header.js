// @flow
import React from "react";
import { View, Text, StyleSheet, TouchableOpacity } from "react-native";
import SafeAreaView from "react-native-safe-area-view";

type HeaderProps = {
  onBackButtonPress: () => void
};

const Header = ({ onBackButtonPress }: HeaderProps) => (
  <SafeAreaView>
    <View style={styles.header}>
      <TouchableOpacity onPress={onBackButtonPress}>
        <Text>Back</Text>
      </TouchableOpacity>
    </View>
  </SafeAreaView>
);

const styles = StyleSheet.create({
  header: {
    height: 180,
    padding: 22
  }
});

export default Header;
