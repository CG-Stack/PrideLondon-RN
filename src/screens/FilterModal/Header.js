// @flow
import React from "react";
import ActionButton from "../../components/ActionButton";
import GlobalHeader from "../../components/Header";
import text from "../../constants/text";

type Props = {
  onClearPress: () => void,
  onCancelPress: () => void,
  showClear: boolean
};

const Header = ({ onClearPress, onCancelPress, showClear }: Props) => {
  const clearElement = showClear ? (
    <ActionButton label={text.clearAll} onPress={onClearPress} />
  ) : null;
  return (
    <GlobalHeader
      leftElement={<ActionButton label={text.cancel} onPress={onCancelPress} />}
      title={text.filterEvents}
      rightElement={clearElement}
    />
  );
};

export default Header;
