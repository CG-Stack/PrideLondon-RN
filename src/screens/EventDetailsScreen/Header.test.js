// @flow
import React from "react";
import { shallow } from "enzyme";
import Header from "./Header";
import Touchable from "../../components/Touchable";

it("renders correctly", () => {
  const output = shallow(
    <Header onBackButtonPress={() => {}} imageUrl="https://image.jpg" />
  );
  expect(output).toMatchSnapshot();
});

it("calls callback when pressed", () => {
  const onPress = jest.fn();
  const output = shallow(
    <Header onBackButtonPress={onPress} imageUrl="https://image.jpg" />
  );

  const touchable = output.find(Touchable);
  touchable.simulate("press");

  expect(onPress).toHaveBeenCalled();
});
