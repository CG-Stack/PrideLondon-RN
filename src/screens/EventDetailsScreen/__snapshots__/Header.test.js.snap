// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ImageBackground
  resizeMode="cover"
  source={
    Object {
      "uri": "https://image.jpg",
    }
  }
  style={
    Object {
      "flex": 1,
      "height": 180,
    }
  }
>
  <withOrientation>
    <StatusBar
      animated={true}
      barStyle="default"
      showHideTransition="fade"
    />
    <Touchable
      accessibilityComponentType="button"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      delayPressIn={50}
      onPress={[Function]}
      style={
        Object {
          "padding": 8,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Back
      </Text>
    </Touchable>
  </withOrientation>
</ImageBackground>
`;
